src/pages/PollingPage.tsx

```diff
@@ -67,31 +93,61 @@ const PollingPage: React.FC = () => {
     setTotalPolls([...deletePoll]);
     setShowDeleteModal(false);
   };
-
-  const handleVote = (itemTitle: string) => {
-    setVotedItems((prevVotedItems) => [...prevVotedItems, itemTitle]);
+  const handleVote = (optionId: number) => {
+    setVotedItems(optionId);
   };
   const handleSubmitVote = (pollTitle: string) => {
-    if (votedItems.length === 1) {
-      setHasVotedMap((prevHasVotedMap) => ({
-        ...prevHasVotedMap,
-        [pollTitle]: true,
-      }));
-    } else if (votedItems.length > 1) {
+    if (votedItems) {
       setHasVotedMap((prevHasVotedMap) => ({
         ...prevHasVotedMap,
         [pollTitle]: true,
       }));
+      dispatch(updateVote(votedItems));
+      localStorage.setItem(
+        "VotedItems",
+        JSON.stringify({ ...hasVotedMap, [pollTitle]: true })
+      );
     } else {
       setHasVotedMap((prevHasVotedMap) => ({
         ...prevHasVotedMap,
         [pollTitle]: false,
       }));
+      localStorage.setItem(
+        "VotedItems",
+        JSON.stringify({ ...hasVotedMap, [pollTitle]: false })
+      );
     }
   };
   const handleEditPoll = (id: number) => {
     navigate(`/editPoll/${id}`);
   };
+  useEffect(() => {
+    const storedVotedItems = localStorage.getItem("VotedItems");
+    if (storedVotedItems) {
+      setHasVotedMap(JSON.parse(storedVotedItems));
+    }
+  }, []);
+
+  const handleViewPollVoteChart = (idOfSelectedPoll: number) => {
+    const dataOfSelectedPoll: any = polls.data.filter((data: any) => {
```

Comment:
why can't you just pass the full data of the clicked Poll? 
you do not need to do filter or anything @Rishab-innovative